### PR TITLE
docker/lite: Fix for other flavors and improved warning output.

### PR DESCRIPTION
**Pull request by michael-berlin**
Follow-up fixes for https://github.com/youtube/vitess/pull/2768

- `make docker_lite` no longer builds the base image before repacking the lite image. @rnavarro wasn't aware of this. This warning should help to clarify things.
- Added `PROMPT_NOTICE` to the other Makefile targets as well.
- `docker/lite/build.sh` now always logs the correct base and lite image if the non-default flavor was selected.

### CODE REVIEW

**Comment by rnavarro** [comment]
I just checked, we weren't affected by this because we use the Percona57 images. That workflow still builds the base image as part of the lite image.

In fact, ALL other workflows still do this.

**Comment by michael-berlin** [comment]
You're right. I totally missed that in the Makefile.

I'll remove it as part of this PR for the sake of consistency.

**Comment by rnavarro** [comment]
As a heads up, I have those changes as part of this PR:

https://github.com/youtube/vitess/pull/2778

**Comment by michael-berlin** [comment]
Thanks for the head's up! I've included the changes in my commit.

**Comment by pivanof** [comment]
The PR contents doesn't match now its title and description. Do you want to modify those or split the PR into different parts?

**Comment by michael-berlin** [comment]
PTAL

I've updated title and description. Changes are related and therefore can stay in one PR.

File: docker/lite/build.sh [hunk 1]
```diff
@@ -14,35 +14,41 @@ fi
...
   tag=$flavor
 else
   echo "Flavor not specified as first argument. Building default image."
-  base_image=vitess/base
 fi
 
 # Abort if base image does not exist.
 if ! docker inspect $base_image &>/dev/null; then
-  echo "ERROR: Dependent image $base_image does not exist. Run 'make docker_base' to build it locally or 'docker pull $base_image' to fetch it from Docker Hub."
+  echo "ERROR: Dependent image $base_image does not exist. Run 'make $make_target' to build it locally or 'docker pull $base_image' to fetch it from Docker Hub (if we publish it)."
```
Comment (pivanof) [inline docker/lite/build.sh]: Probably it's worth saying "if it's published" to not make people guess who is "we" here.

File: docker/lite/build.sh [hunk 2]
```diff
@@ -78,9 +84,9 @@ chmod -R o=g lite
 # Build vitess/lite image
 
 if [[ -n "$flavor" ]]; then
-	docker build --no-cache -f Dockerfile.$flavor -t vitess/lite:$flavor .
+  docker build --no-cache -f Dockerfile.$flavor -t vitess/lite:$flavor .
```
Comment (pivanof) [inline docker/lite/build.sh]: Might be worth putting $lite_image here.

File: docker/lite/build.sh (hunk 1 above)
Comment (michael-berlin) [inline docker/lite/build.sh]: Good idea! I like it.

Done.

File: docker/lite/build.sh (hunk 2 above)
Comment (michael-berlin) [inline docker/lite/build.sh]: Done.

I've put $lite_image here and also defined a variable for `Dockerfile(.$flavor)`. This way I could get rid of the `if` here.

**Pull request merged**